dbt/contracts/connection.py

```diff
@@ -36,6 +36,52 @@
     'required': ['dbname', 'host', 'user', 'pass', 'port', 'schema'],
 }
 
+REDSHIFT_CREDENTIALS_CONTRACT = {
+    'type': 'object',
+    'additionalProperties': False,
+    'properties': {
+        'method': {
+            'enum': ['database', 'iam']
+        },
+        'dbname': {
+            'type': 'string',
+        },
+        'host': {
+            'type': 'string',
+        },
+        'user': {
+            'type': 'string',
+        },
+        'pass': {
+            'type': 'string',
+        },
+        'port': {
+            'oneOf': [
+                {
+                    'type': 'integer',
+                    'minimum': 0,
+                    'maximum': 65535,
+                },
+                {
+                    'type': 'string'
+                },
+            ],
+        },
+        'schema': {
+            'type': 'string',
+        },
+        'cluster_id': {
+            'type': 'string'
+        },
+        'iam_duration_seconds': {
+            'type': ['null', 'integer'],
+            'minimum': 900,
+            'maximum': 3600
```

Comment:
does this mean if a model takes >1hr to run, that connection will close? (that's fine, just want to understand)

Reply:
I think that's right. I couldn't find any specifics in the docs, but keen to set the timeout to 900s then experiment with a `time.sleep()` after acquiring the connection. I bet that once a query is issues, it will complete even if the creds are expired, but subsequent queries on the same connection will likely fail.

I wish we could make the timeout longer, but 3600s is the max per the [docs](https://docs.aws.amazon.com/redshift/latest/APIReference/API_GetClusterCredentials.html)

Reply:
yeah this is pretty typical of aws federated auth. >1hr would be great.